File: cf-1327-b.py
#                          JAI SHREE RAM

import math; from collections import *
import sys; from functools import reduce

# sys.setrecursionlimit(10**6)
def get_ints(): return map(int, input().strip().split())
def get_list(): return list(get_ints())
def get_string(): return list(input().strip().split())
def printxsp(*args): return print(*args, end="")
def printsp(*args): return print(*args, end=" ")

UGLYMOD = int(1e9)+7; SEXYMOD = 998244353; MAXN = int(1e5)

# sys.stdin=open("input.txt","r");sys.stdout=open("output.txt","w")

for _testcases_ in range(int(input())):
    n = int(input())
    ans = -1
    visited = [False] * (n+1)
    for i in range(1, n+1):
        li = get_list()[1:]
        for l in li:
            if not visited[l]:
                visited[l] = True
                break
        else:
            ans = i
    if ans != -1:
        v = visited.index(False, 1, n+1)
    print("IMPROVE\n{} {}".format(ans, v) if ans != -1 else "OPTIMAL")
        


'''
>>> COMMENT THE STDIN!! CHANGE ONLINE JUDGE !!
THE LOGIC AND APPROACH IS MINE @luctivud ( UDIT GUPTA )
Link may be copy-pasted here if it's taken from other source.
DO NOT PLAGIARISE.
>>> COMMENT THE STDIN!! CHANGE ONLINE JUDGE !!
'''

File: cf-1327-b.cpp
//                           जय श्री राम

#include <bits/stdc++.h>
using namespace std;

const long long int UGLYMOD {1000000007};
const long long int SEXYMOD {998244353};
const long int MAXN = (long int) 1e6;

typedef long long int ll;

#define LUCILLE_IS_THIRSTY()       ios_base::sync_with_stdio(false); cin.tie(0)
#define TESTCASES_ARE_THERE()      long int Test; cin>>Test; while(Test--)
#define mems(arr,val)              memset((arr), val, sizeof((arr)))
#define endl                       "\n"

inline int readint() {
    int res = 0; char c = 0;
    while(!isdigit(c)) c = getchar();
    while(isdigit(c)) res = res*10+c-'0', c = getchar();
    return res;
}


void solveEach() {
    long int n;
    scanf("%ld", &n);
    bool visited[n+1] = {false};
    long int ans = -1;
    for (long int i = 1L; i<n+1; i++) {
        long int x; 
        scanf("%ld", &x);
        vector<long int> arr(x);
        for(long int j = 0; j<x; j++) {
            scanf("%ld", &arr[j]);
        }
        bool flag2 = true;
        for (auto j: arr) {
            if (!visited[j]) {
                visited[j] = true;
                flag2 = false;
                break;
            }
        }
        if (flag2) {
            ans = i;
        }
        arr.clear();
    }
    long int ans2 = -1;
    if (ans != -1) {
        for (long int k=1; k<n+1; k++) {
            if (!visited[k]){
                printf("IMPROVE\n%ld %ld\n", ans, k);
                break;
            }
        }
    } else {
        printf("OPTIMAL\n");
    }
    
    return;
}

signed main() {

    // LUCILLE_IS_THIRSTY();

    #ifndef ONLINE_JUDGE
        freopen("input.txt", "r", stdin);
        freopen("output.txt", "w", stdout);
    #endif 

    TESTCASES_ARE_THERE()
        solveEach();
    return 0;
}
/*
THE LOGIC AND APPROACH IS BY ME @luctivud ( UDIT GUPTA )
SOME PARTS OF THE CODE HAS BEEN TAKEN FROM WEBSITES LIKE::
(I Own the code if no link is provided here or I may have missed mentioning it)
>>> DO NOT PLAGIARISE.
TESTCASES:
*/